app/controllers/chess_controller.rb

```diff
@@ -1,7 +1,17 @@
 class ChessController < ApplicationController
   def index
+    @games = Game.all
+    @my_games = []
+    @finished_games = []
+    if current_user
+      @games.each do |game|
+        @my_games << game if (game.white_player == current_user.id || game.black_player == current_user.id) && game.finished == false
```

Comment:
You can reduce the cyclomatic complexity by moving some of these conditions into a new method. Also, I think it makes more sense to have finished games only show up in the finished games section, and not the full games section.

        @my_games << game if game_player(game) && !game.finished
        @finished_games << game if game_player(game) && game.finished
      end
    end
    @available_games = Game.available
    @full_games = @games - @available_games - @finished_games
    flash[:notice] = nil
    flash[:alert] = nil
  end

  private

  def game_player(game)
    (game.white_player == current_user.id || game.black_player == current_user.id)
  end